internal/equality/equality.go

```diff
@@ -222,6 +222,11 @@ func LoadBalancerServiceChanged(current, expected *corev1.Service) (*corev1.Serv
 		changed = true
 	}
 
+	if !apiequality.Semantic.DeepEqual(current.Spec.LoadBalancerIP, expected.Spec.LoadBalancerIP) {
+		updated.Spec.LoadBalancerIP = expected.Spec.LoadBalancerIP
+		changed = true
+	}
+
```

Comment:
This will cause Contours that do not specify a Service IP to fail the equality comparison. I suggest either passing in a `*operatorv1alpha1.Contour` to the func so this check is only performed on Contour objects that have a static IP assigned for provider params.

Reply:
I did some testing (adding, removing `address` parameter, swapping between providers etc.) and I am not getting any errors. To me it seems that in case we do not provide `address` parameter in Contour, it will consider this as empty and will not try to add it.
Maybe I am just looking at wrong place - can you maybe point me in the right direction? Also how would you add the check for `address` parameter in this case?

Reply:
@BostjanBozic one option is to update the function's signature to require a *Contour and then only call your new code if the Contour contains any static IP params introduced by this PR.

For example:
```
func LoadBalancerServiceChanged(current, expected *corev1.Service, contour *operatorv1alpha1.Contour) (*corev1.Service, bool) {
   ...
   if contour.AWS|GCP|Azure provider params specify an IP then run your equality check.
}
```
You then need to update callers of `LoadBalancerServiceChanged` to include a *Contour.